src/serac/numerics/dense_petsc.hpp

```diff
@@ -16,7 +16,13 @@ struct DenseMat {
 
   /// @brief constructor
   /// @param a matrix
-  DenseMat(const DenseMat& a) { MatDuplicate(a.A, MAT_COPY_VALUES, &A); }
+  DenseMat(const DenseMat& a)
+  {
+    MatDuplicate(a.A, MAT_COPY_VALUES, &A);
+    MatCopy(a.A, A, SAME_NONZERO_PATTERN);
+    MatAssemblyBegin(A, MAT_FINAL_ASSEMBLY);
+    MatAssemblyEnd(A, MAT_FINAL_ASSEMBLY);
+  }
```

Comment:
You've done the testing, so I trust you on this. If you actually wanted me to double check, let me know. Otherwise mark as resolved.

Reply:
The assembly are definitely needed.  The Copy, maybe not.  But I'm sick of trying to guess what PETSc is going to do, best be safe.  It costs nothing in practice.